torch/nn/functional.py

```diff
@@ -705,6 +707,85 @@ def embedding(input, embedding_matrix,
+                                 as the `input` is treated as a mini-batch of fixed length sequences of length `N` each.
+            max_norm (float, optional): If given, will renormalize the embeddings to always have a norm lesser than this
+            norm_type (float, optional): The p of the p-norm to compute for the max_norm option
+            scale_grad_by_freq (boolean, optional): if given, this will scale gradients by the frequency of
+                                                    the words in the dictionary.
+            mode (string, optional): 'sum' | 'mean'. Specifies the way to reduce the bag. Default: 'mean'
+
+        Shape:
+            - Embedding_matrix: FloatTensor `(V, embedding_dim)`,
+                                V = number of embeddings, embedding_dim = embedding size
+            - Input: LongTensor `N`, N = number of embeddings to extract
+                     (or) LongTensor `BxN`, B = number of sequences in mini-batch,
+                                            N = number of embeddings per sequence
+            - Offsets: LongTensor `B`, B = number of bags. The values are the
+                       offsets in `input` for each bag, i.e. the cumsum of lengths.
+                       Offsets is not given if Input is 2D `BxN` Tensor,
+                       the input is considered to be of fixed-length sequences
+            - Output: `(B, embedding_dim)`
+
+        Examples::
+
+            >>> # an Embedding module containing 10 tensors of size 3
+            >>> embedding_matrix = Variable(torch.rand(10, 3))
+            >>> # a batch of 2 samples of 4 indices each
+            >>> input = Variable(torch.LongTensor([1,2,4,5,4,3,2,9]))
+            >>> offsets = Variable(torch.LongTensor([0,4]))
+            >>> embedding_bag(embedding_matrix, input, offsets)
+
+            Variable containing:
+            -1.1840 -0.2547 -0.5860
+            -0.7126  0.0002 -0.3411
+            [torch.FloatTensor of size 2x3]
+
+        """
+    if indices.dim() == 2:
+        if offsets is not None:
+            raise ValueError("if input is 2D, then offsets has to be None"
+                             ", as input is treated is a mini-batch of"
+                             " fixed length sequences. However, found "
+                             "offsets of type {}".format(type(offsets)))
+        else:
+            offsets = Variable(torch.arange(0, indices.numel(), indices.size(1),
+                               out=indices.data.new().long()))
+            indices = indices.view(-1)
+
+    elif indices.dim() != 1:
+        raise ValueError("input has to be 1D or 2D Tensor,"
+                         " but got Tensor of dimension {}".format(indices.dim()))
+
+    if offsets is None:
+        raise ValueError("offsets has to be a 1D Tensor but got None")
+
+    return torch.nn.backends.thnn.backend.EmbeddingBag.apply(
```

Comment:
It's simpler to do `_functions.thnn.EmbeddingBag`

Reply:
sorry, I got confused since `Embedding` also calls its function via `torch.nn`.

Reply:
No worries. I think this is valid too, it's just unnecessarily long